packages/beacon-node/src/network/processor/gossipHandlers.ts

```diff
@@ -279,7 +280,7 @@ function getDefaultHandlers(modules: ValidatorFnsModules, options: GossipHandler
           }
         }
         metrics?.gossipBlock.processBlockErrors.inc({error: e instanceof BlockError ? e.type.code : "NOT_BLOCK_ERROR"});
-        logger.error("Error receiving block", {slot: signedBlock.message.slot, peer: peerIdStr}, e as Error);
+        logger.warn("Error receiving block", {slot: signedBlock.message.slot, peer: peerIdStr}, e as Error);
```

Comment:
We could also consider `debug` here but these logs are really rare and might indicate an issue with the network as a whole, e.g. if there is an issue in another client producing invalid blocks. Also see Lion's comment on this previously https://github.com/ChainSafe/lodestar/pull/5457#discussion_r1184514672.

Reply:
its better to debug/warn/error in case of conditions that we see, and would by default error so that its very visible unless its a scenario we have analyzed and hence put it to debug/warn

Reply:
I do agree with that, we should be very careful with moving log to less verbosity but this is a clear case to me as the log is triggered based on data we receive from the network, so at most is should be a warning as the user / operator cannot be expected to do something about it.

It is also quite clear based on the definition of log level in our logging policy
![image](https://github.com/ChainSafe/lodestar/assets/38436224/f778c902-ce11-46a5-ad5e-277cedc96cfb)


What about we adapt the log level (verbosity) based on the error we receive here?
- unknown error (not a BlockError) = `error`
- `e instanceof BlockError` = `warn`
- `BlockErrorCode.DATA_UNAVAILABLE` = `debug`